### DISCUSSION TITLE
Log data (to flash, or to SD)

### DISCUSSION THREAD

**Discussion by renatobo** [renatobo/bonogps#58] [Ideas]
Simple option: log data to flash (this might require removing the option for OTA)
Additional HW needed for the option to log to SD (probably the better option)

Examples at

* https://randomnerdtutorials.com/esp32-data-logging-temperature-to-microsd-card/
* https://github.com/wilhelmzeuschner/esp32_gps_bicycle_computer
* https://github.com/har-in-air/ESP32_IMU_BARO_GPS_LOGGER
* https://github.com/hamsternz/esp32_serial_log_to_sd

**Comment by Ralphy140** [comment]
I have followed your project and just finished the build and confirmed working with Harry's on IOS. 
I would love to see this feature implemented to write out a GPX file as it would remove the IOS compatibility challenges, I have just ordered some micro SD boards to have a test with. I will report back on my progress though i am not a programmer so i am sure it will be slow

**Comment by renatobo** [comment]
I would suggest logging a binary format (to EPROM FS or to SD) and then change into GPX file when exporting.

You need to parse the GPS responses into position data (all the interfaces here implemented only transfer data "as-is" and the mobile phone apps translate that into their own format): there is some code already in place that uses the [NeoGPS](https://github.com/SlashDevin/NeoGPS) library.
See code enabled via macro NEED_NEOGPS

One recommendation would be: configure the GPS output to use binary ublox format and reduce the output to position only (no equivalent of GSA/GSV), this will minimize serial transmission between esp32 and data parsing.
